Transfer to the new repo.

LightnessOfBeing opened the pull request.

LightnessOfBeing committed:
Transfer to the new repo.

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+ * Class that implements trie.
+ */
+
+public class Trie implements Serializable{
```
Перед "{" нужен пробел, здесь и ниже

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+        }
+
+        public TrieNode get(char c){
+            if(mapChild == null){
```
И перед "(" нужен пробел

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+     * Size of a trie.
+     */
+
+    private int size = 0;
```
Подозреваю, что size всегда равен root.howManyWords

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+
+        private int howManyWords;
+
+        TrieNode(Character c){
```
Оно должно быть либо public, либо private (ужасно, что в Java и то и другое будет работать)

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+        root = new TrieNode((char) 0);
+    }
+
+    public int size(){
```
Нужен комментарий

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+     */
+    public boolean contains(String element){
+        TrieNode node = root;
+        for (char c : element.toCharArray()){
```
Так оно упадёт, если element null

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+            cnt++;
+        }
+        if(node.getMap() != null) {
+            HashMap<Character, TrieNode> m = (HashMap<Character, TrieNode>) node.getMap();
```
А зачем этот каст? Зачем m быть именно HashMap-ом, а не более абстрактным интерфейсом Map?

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+            HashMap<Character, TrieNode> m = (HashMap<Character, TrieNode>) node.getMap();
+            for (Character a : m.keySet()) {
+                cnt += m.get(a).getHowManyWords();
+            }
```
А это разве не просто node.getHowManyWords? И если нет, то почему? :)

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+            }
+
+        } catch(IOException i){
+            i.printStackTrace();
```
Не надо его ловить, чтобы распечатать стек. Может, пользователь не хочет стек на консоли. И раз Вы его тут обработали, зачем `throws IOException`? Правильнее просто дать ему упасть

yurii-litvinov commented on TrieHW4/src/main/java/ru/spbau/mit/vishnyakov/Trie.java:
```diff
@@ -0,0 +1,288 @@
+     * Serializes our trie.
+     * We start from the root and serialize node by node via dfs.
+     * @param out stream where we want to write.
+     * @throws IOException
```
Тут положено писать, в каких ситуациях исключение бросается, либо не писать тэг вообще

yurii-litvinov commented on TrieHW4/src/test/java/ru/spbau/mit/vishnyakov/TrieTest.java:
```diff
@@ -0,0 +1,454 @@
+        Trie t = new Trie();
+
+        t.add("a");
+        assertEquals(true, t.contains("a"));
```
Тут лучше assertTrue

yurii-litvinov commented on TrieHW4/src/test/java/ru/spbau/mit/vishnyakov/TrieTest.java:
```diff
@@ -0,0 +1,454 @@
+
+        t.remove("aaa");
+
+        assertEquals(false, t.contains("aaa"));
```
А тут --- assertFalse

yurii-litvinov commented on TrieHW4/src/test/java/ru/spbau/mit/vishnyakov/TrieTest.java:
```diff
@@ -0,0 +1,454 @@
+    @Test
+    public void deserialize() throws IOException, ClassNotFoundException {
+        Trie t = new Trie();
+        InputStream in = new FileInputStream("out.txt");
```
Если этот тест запустить отдельно, он не пройдёт. А тесты должны быть независимы, поскольку они могут запускаться по отдельности, в произвольных комбинациях и в произвольном порядке. Да, и тесты должны подчищать за собой временные файлы.

yurii-litvinov reviewed (changes requested):
Оу, а зачем в новом репозитории бинарники и сгенерённые файлы? https://github.com/github/gitignore/blob/master/Java.gitignore, https://github.com/github/gitignore/blob/master/Gradle.gitignore и https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore, если их померджить, решат проблему.

LightnessOfBeing committed:
Updated Trie

Removed binary files, fixed all style complaints, added test which handles incorrect input element. Reworked remove method, howManyStartsWithPrefix and add methods. Fixed all  tests. Also added serializes.txt and deserialize.txt which are being used for testing of serialize and deserialize method. These are not temporary files

yurii-litvinov commented on TrieHW4/build/reports/tests/test/css/style.css:
```diff
@@ -0,0 +1,84 @@
+
```
Вот это не надо

yurii-litvinov commented on TrieHW4/build/reports/tests/test/index.html:
```diff
@@ -0,0 +1,132 @@
+<!DOCTYPE html>
```
И это

yurii-litvinov commented on TrieHW4/build/reports/tests/test/js/report.js:
```diff
@@ -0,0 +1,194 @@
+(function (window, document) {
```
И это

yurii-litvinov commented on TrieHW4/build/reports/tests/test/packages/ru.spbau.mit.vishnyakov.html:
```diff
@@ -0,0 +1,103 @@
+<!DOCTYPE html>
```
И это

yurii-litvinov commented on TrieHW4/build/test-results/test/TEST-ru.spbau.mit.vishnyakov.TrieTest.xml:
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
И это

yurii-litvinov reviewed (changes requested):
Всю папку build надо убрать

LightnessOfBeing committed:
Deleted build folder.

yurii-litvinov reviewed (approved):
👍
